clouddrift/analysis.py

```diff
@@ -103,6 +103,7 @@ def chunk(
     x: Union[list, np.ndarray, xr.DataArray, pd.Series],
     length: int,
     overlap: int = 0,
+    keep: str = "mid",
```

Comment:
I think the name `keep` is confusing here. To keep the points implies that the purpose of the function is to discard points. However, discarding is merely a consequence of the input not matching the size of the output.

What do you think about naming this argument `shift` or `align`, with possible options being "left", "center", "right"?

Reply:
Alternatively, `discard` or `trim` (meaning, opposite of `keep`) would work well too.

Reply:
Actually `keep=mid` does keeps the middle point (discard if needed start and end). Similarly, with `keep=start`, chunks will start at the beginning.

Reply:
Because of the align to the middle behavior, I realize my second suggestion (`discard` or `trim`) doesn't work well.

I understand the meaning of `keep`, and it makes sense when I think through it. I just personally found it a bit more difficult to understand. Because the function is called `chunk`, and it chunks an input array into a 2-d array of equal-length chunks, the argument `keep` I couldn't help but ask "keep what?". It's not "keep the chunk", but something else. Then I had to read the description. If we call this `shift` or `align`, the verb naturally addresses the chunks, so I find it a bit more intuitive. Maybe it's just my brain.

Reply:
I think that `keep` and `align` are both good but `left` and `right` as possible options not so much. I visualize a ragged array as being a column vector, not a row vector, but others may see it differently. As such, I prefer `(start,[mid,center],end)` to `(left,center,right)`.

Reply:
That's a good point. I suggest `mid` -> `middle` for consistency with `start` and `end` (they are not abbreviated).

Reply:
and we use "keep" or "align"?

Reply:
The goal of the function is to create the chunks so let's make this option about the output, i.e. how do we want the output to be aligned with respect to the input. So let's use `align`.